### DISCUSSION TITLE
download without any ending

### DISCUSSION THREAD

**Discussion by gpahlevanzadeh** [k3s-io/k3s#11495] [Q&A] [answered]
I want to install agent such as:
`root@agent1:~# curl -sfL https://get.k3s.io | K3S_TOKEN=mohsenjoojoo sh -s - agent --server https://192.168.1.106:6443`

But it stay on binary download such as:
```
[INFO]  Finding release for channel stable
[INFO]  Using v1.31.4+k3s1 as release
[INFO]  Downloading hash https://github.com/k3s-io/k3s/releases/download/v1.31.4+k3s1/sha256sum-amd64.txt
[INFO]  Downloading binary https://github.com/k3s-io/k3s/releases/download/v1.31.4+k3s1/k3s
```
I get query with `du` in download duration I get the following messaes:
```
root@agent1:~# du /tmp/k3s-install.0tau6Oc9Ma/k3s.bin 
16172	/tmp/k3s-install.0tau6Oc9Ma/k3s.bin
root@agent1:~# du /tmp/k3s-install.0tau6Oc9Ma/k3s.bin 
16204	/tmp/k3s-install.0tau6Oc9Ma/k3s.bin
root@agent1:~# du /tmp/k3s-install.0tau6Oc9Ma/k3s.bin 
16316	/tmp/k3s-install.0tau6Oc9Ma/k3s.bin
```
Result of `du` command say downloader is downloading but it act such as loop. 
How can I install agent without this problem? Is it a BUG?

**Comment by brandond** [comment]
The install script downloads artifacts from github with curl or wget. You can add -x to the flags to have it print the commands while it's running them:

`curl -sfL https://get.k3s.io | K3S_TOKEN=mohsenjoojoo sh -x -s - agent --server https://192.168.1.106:6443`

You'll need to figure out on your own why your node cannot access github though.

**Reply by gpahlevanzadeh** [reply, marked as answer]
Your network has problem.
